lib/elixir/lib/kernel.ex

```diff
@@ -795,11 +798,13 @@ defmodule Kernel do
 
       iex> trunc(5.4)
       5
+
       iex> trunc(5.99)
       5
 
   """
-  @spec trunc(number) :: integer
+  @spec trunc(arg) :: arg when arg: integer
```

Comment:
Yeah I think it's just a case of the funky docs™, I interpreted it same as @ericmj said. Then since this is the case, `@spec trunc(number) :: integer` is perfectly fine.